include/common/bitdepth.rs

```diff
@@ -193,6 +204,26 @@ pub trait BitDepth: Clone + Copy {
 
     fn get_intermediate_bits(&self) -> u8;
 
+    fn cast_pixel_slice(bytes: &[u8]) -> &[Self::Pixel] {
+        let len = Self::pxstride(bytes.len());
+
+        assert!(bytes.len() % len == 0);
```

Comment:
```suggestion
        assert!(len * mem::size_of::<Self::Pixel>() <= bytes.len());
```

Shoudn't we check the length is in bounds?

Reply:
Wait now that I think about it that explanation isn't correct, because we're also changing the type of the pointer. I'm pretty sure we don't need to check the length explicitly, but maybe I should change the logic to use `size_of::<Self::Pixel>` instead of pxstride 🤔

Reply:
Okay I've rewritten it to use `size_of`, which I think makes it clearer that we're checking the length correctly without relying on the behavior of `pxstride`.